### DISCUSSION TITLE
Challenge 5 help needed

### DISCUSSION THREAD

**Discussion by 0xtdey** [Cyfrin/foundry-full-course-cu#215] [Q&A] [answered]
I have tried solving the challenge, as keccak256 hashing function is one way function and it cannot be reversed. I tried using AI to come up with code in python which used the hashlib library to find the matching input string, and in C++ with openSSL lib to do the same. 

Still no luck. 

Any help or hint would be really appreciated!

**Comment by lamtrinh259** [comment, marked as answer]
@tamoghna-dey I just solved it! Reversing it is entirely impossible and would take an infinite amount of time. 

I suggest trying with some very common words and then you may be able to discover it. Another hint: since you're using AI to write the code to test out, try to compile a list of commonly compromised passwords (i.e., when Google tells you that your password is unsafe) and loop them in your AI and then you will discover the answer.

**Reply by 0xtdey** [reply]
Thanks a lot you perfectly gave the hint, I solved it in one go!
